tests/jobs/test_backfill_job.py

```diff
@@ -719,7 +748,7 @@ def test_backfill_retry_always_failed_task(self, dag_maker):
             start_date=DEFAULT_DATE,
             end_date=DEFAULT_DATE,
         )
-        with pytest.raises(BackfillUnfinished):
+        with self.assertRaises(BackfillUnfinished):
```

Comment:
```suggestion
        with pytest.raises((BackfillUnfinished):
```
We no longer use unittest model on these tests, so this won't work